main.cpp

```diff
@@ -31,9 +31,145 @@
 
  Wait for my code review.
  */
-
 #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs);
+    float subtract(float lhs, float rhs);
+    float multiply(float lhs, float rhs);
+    float divide(float lhs, float rhs);
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs)
+{
+    if (rhs == 0.0f)
+    {
+        std::cout << "Error: cannot divide by 0! Returning 0 instead\n";
+        return 0;
+    }
+    else
+        return lhs / rhs;
+}
+
+struct DoubleType
+{
+    double add(double lhs, double rhs);
+    double subtract(double lhs, double rhs);
+    double multiply(double lhs, double rhs);
+    double divide(double lhs, double rhs);
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{
+    return lhs - rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+
+double DoubleType::divide(double lhs, double rhs)
+{
+    if (rhs == 0.0)
+    {
+        std::cout << "Error: cannot divide by 0! Returning 0 instead\n";
+        return 0;
+    }
+    else
+        return lhs / rhs;
+}
+
+struct IntType
+{
+    int add(int lhs, int rhs);
+    int subtract(int lhs, int rhs);
+    int multiply(int lhs, int rhs);
+    int divide(int lhs, int rhs);
+};
+
+int IntType::add(int lhs, int rhs)
+{
+    return lhs + rhs;
+}
+
+int IntType::subtract(int lhs, int rhs)
+{
+    return lhs - rhs;
+}
+
+int IntType::multiply(int lhs, int rhs)
+{
+    return lhs * rhs;
+}
+
+int IntType::divide(int lhs, int rhs)
+{
+    if (rhs == 0)
+    {
+        std::cout << "Error: cannot divide by 0! Returning 0 instead\n";
+        return 0;
+    }
+    else
+        return lhs / rhs;
```

Comment:
If you're gonna use `{ }` in an `if/else`, you should use it for both of them.  

- [ ] formatting